File: Programmation/TD5/meteo.pl
#!/usr/bin/env perl
use strict ;
use warnings ;

open(METEO, '<', 'temperatures.csv') ;

my %meteo ;
while (<METEO>) {
  if (/(\w+) (\d+) (\d+\.\d+)/) {
    $meteo{"$1"} = "$3" ;
  }
}

my @sorted_keys = sort { $meteo{$b} <=> $meteo{$a} } keys %meteo;
print "La ville la plus chaude est $sorted_keys[0] avec $meteo{$sorted_keys[0]}°C de moyenne\n";
print "À l'inverse, la ville la plus froide est $sorted_keys[-1] avec $meteo{$sorted_keys[-1]}°C de moyenne\n";

File: Programmation/TD5/meteo.rb
#!/usr/bin/env ruby

meteo = Hash.new

File.open("temperatures.csv").each do |line|
  if (/(\w+) (\d+) (\d+.\d+)/.match(line)) 
    meteo[$1] = $3.to_f
  end
end

puts "La ville la plus chaude est #{meteo.index(meteo.values.max)} avec #{meteo.values.max}°C de moyenne
À l'inverse, la ville la plus froide est #{meteo.index(meteo.values.min)} avec #{meteo.values.min}°C de moyenne"
